next/pages/vyhladavanie.tsx

```diff
@@ -62,9 +54,10 @@ const Search = ({ footer, mainMenu, page }: AsyncServerProps<typeof getServerSid
   const [searchQuery, setSearchQuery] = useState<string>(debouncedSearchInputValue)
 
   useEffect(() => {
-    if (debouncedSearchInputValue.length > minKeywordLength) {
+    if (debouncedSearchInputValue !== searchQuery) {
       setSearchQuery(debouncedSearchInputValue)
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [debouncedSearchInputValue])
```

Comment:
are you sure (here and on the other place as well) - it seems to me you'd want this to change with searchQuery as well, otherwise you may be comparing against incorrect value

Reply:
I think there should not be a problem with it for now, because the change should happen alway in one way from debounced value to search value. 
But I'll examine it properly, I would like to merge this PR and make changes in further PR if needed.